# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/smibrahimali/Flyrank-Intern/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [7]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

Lane Choice: Lane 2 — Refresh / Content Opportunity Scoring.

Why: Content teams have finite time and budget; they cannot rewrite or refresh every aging article. This lane treats machine learning as an operational decision-support tool. It prioritizes which decaying pages offer the highest return on editorial effort, filtering noise before human intervention.

Total pages in starter slice: 30,000
Pages showing downward trend: 16,262 (54.2%)
High-value declining pages (>= 500 impressions): 9,961 (33.2%)


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [4]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
The Search Question: Which underperforming or decaying pages should be reviewed first for a content refresh, based on historical demand, current visibility, and engagement signals?
Unit of Analysis: One row per pseudonymized content item (page) per client.
The Output: A ranked queue of candidate pages with confidence scores and human-readable reason codes (e.g., declining_with_demand, ctr_review_candidate).
The Action: An editorial reviewer inspects the top-ranked candidates and decides whether to update, expand, restructure, or monitor the page.
Cost of a Wrong Call:
False Positive: The editorial team wastes hours rewriting a page that was merely experiencing low-volume noise or expected seasonality.
False Negative: High-value pages quietly lose traffic unnoticed, leading to a sustained drop in organic visibility.
Why Data/ML Helps: Fixed rules (e.g., "age > 180 days") produce noisy queues with low precision. A learned model can weigh interactions across position tiers, impression volume, and engagement rate gaps to rank the most actionable pages first.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [8]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

# Load the dataset (assuming you already ran !git clone earlier in your session)
data_path = 'Flyrank-Intern/data/raw/content_refresh_anonymized.csv'
df = pd.read_csv(data_path)

total_pages = len(df)
declining_pages = len(df[df['trend_direction'] == 'down'])
high_value_decline = len(df[(df['trend_direction'] == 'down') & (df['impressions_90d'] >= 500)])

print(f"Total pages in starter slice: {total_pages:,}")
print(f"Pages showing downward trend: {declining_pages:,} ({(declining_pages/total_pages)*100:.1f}%)")
print(f"High-value declining pages (>= 500 impressions): {high_value_decline:,} ({(high_value_decline/total_pages)*100:.1f}%)")

Total pages in starter slice: 30,000
Pages showing downward trend: 16,262 (54.2%)
High-value declining pages (>= 500 impressions): 9,961 (33.2%)


Data Takeaway:
The dataset contains 30,000 pages. While 16,262 (54.2%) display a downward trend, only 9,961 (33.2%) meet the minimum volume threshold of at least 500 impressions over 90 days. This demonstrates why simple heuristic flags produce too much noise, and why an opportunity scoring model is needed to prioritize the subset that actually impacts business results.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [6]:
What can be claimed: The model identifies observational patterns and associations between search/engagement metrics and content decay to produce a prioritized decision-support queue.
What cannot be claimed: We cannot claim causality (i.e., updating the page will guarantee traffic recovery), nor does this reverse-engineer Google's proprietary search ranking algorithms.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.